vxquery-rest/src/test/java/org/apache/vxquery/rest/RestServerTest.java

```diff
@@ -0,0 +1,216 @@
+                                                 "<title>Proceedings of the 1990 ACM SIGMOD International Conference on Management of Data, Atlantic City, NJ, May 23-25, 1990.</title>\n";
+
+    private static VXQueryApplication application;
+    private static RestServer restServer;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        System.setProperty(Constants.Properties.VXQUERY_PROPERTIES_FILE, "src/test/resources/vxquery.properties");
+        application = new VXQueryApplication();
+        application.start();
+
+        restServer = application.getRestServer();
+    }
+
+    @Test
+    public void testJSONResponses() throws Exception {
+        runTest(CONTENT_TYPE_JSON);
+    }
+
+    @Test
+    public void testXMLResponses() throws Exception {
+        runTest(CONTENT_TYPE_XML);
+    }
+
+    private void runTest(String contentType) throws Exception {
+        URI queryEndpointUri = new URIBuilder()
+                                       .setScheme("http")
+                                       .setHost("localhost")
+                                       .setPort(restServer.getPort())
+                                       .setPath(QUERY_ENDPOINT)
+                                       .addParameter(STATEMENT, QUERY)
+                                       .addParameter(SHOW_AST, "true")
+                                       .addParameter(SHOW_TET, "true")
+                                       .addParameter(SHOW_OET, "true")
+                                       .addParameter(SHOW_RP, "true")
+                                       .build();
+
+        QueryResponse queryResponse = getQueryResponse(queryEndpointUri, contentType);
+        Assert.assertNotNull(queryResponse);
+
+        Assert.assertEquals(Status.SUCCESS.toString(), queryResponse.getStatus());
+        Assert.assertNotNull(queryResponse.getResultId());
+        Assert.assertNotNull(queryResponse.getRequestId());
+        Assert.assertNotNull(queryResponse.getResultUrl());
+        Assert.assertNotNull(queryResponse.getAbstractSyntaxTree());
+        Assert.assertNotNull(queryResponse.getTranslatedExpressionTree());
+        Assert.assertNotNull(queryResponse.getOptimizedExpressionTree());
+        Assert.assertNotNull(queryResponse.getRuntimePlan());
+
+        QueryResultResponse resultResponse = getQueryResultResponse(queryResponse.getResultId(), contentType);
+        Assert.assertNotNull(resultResponse);
+        Assert.assertNotNull(resultResponse.getStatus());
+
+        Assert.assertNotNull(resultResponse.getResults());
+        Assert.assertNotNull(resultResponse.getRequestId());
+        Assert.assertEquals(RESULT.replace("\n", ""), resultResponse.getResults().replace("\n", ""));
+
+    }
+
+    private static QueryResponse getQueryResponse(URI uri, String accepts) throws IOException, JAXBException {
```

Comment:
That was in my mind to write more test cases for different queries. I will be able to work on that this week.